core/src/replay_stage.rs

```diff
@@ -1390,15 +1390,21 @@ impl ReplayStage {
     ) -> (ProgressMap, HeaviestSubtreeForkChoice) {
         let (root_bank, frozen_banks, duplicate_slot_hashes) = {
             let bank_forks = bank_forks.read().unwrap();
+            let root_bank = bank_forks.root_bank();
             let duplicate_slots = blockstore
-                .duplicate_slots_iterator(bank_forks.root_bank().slot())
+                // It is important that the root bank is not marked as duplicate on initialization.
+                // Although this bank could contain a duplicate proof, the fact that it was rooted
+                // either during a previous run or artificially means that we should ignore any
+                // duplicate proofs for the root slot, thus we start consuming duplicate proofs
+                // from the root slot + 1
+                .duplicate_slots_iterator(root_bank.slot().saturating_add(1))
```

Comment:
Wonder if we should be filtering out duplicate confirmed blocks here, or will replay naturally take care of them.

We only count landed votes toward duplicate confirmed so sounds like replay should cover this

Reply:
replay should take care of any duplicate confirmed past the root bank. although they will be marked as duplicate initially on restart, once we replay the votes they will receive the same duplicate confirmed status they had before.

It seems like the duplicate confirmed column is only used for ancestor hashes repair? so there might be a temporary mismatch where we serve ancestor hashes repair while our local fork choice still has the slot marked as duplicate.